saleor/graphql/shipping/tests/test_shipping_zones_query.py

```diff
@@ -308,3 +308,32 @@ def test_query_shipping_zone_search_by_channels(
 
     assert data[0]["node"]["name"] == shipping_zone_usd.name
     assert data[0]["node"]["id"] == shipping_zone_usd_id
+
+
+def test_query_shipping_zone_search_by_channels_no_matter_of_input(
+    staff_api_client,
+    shipping_zone_with_different_channels,
+    permission_manage_shipping,
+    channel_USD,
+    channel_PLN,
+):
+    # given
+    shipping_zones = shipping_zone_with_different_channels
+    channel_id = graphene.Node.to_global_id("Channel", channel_PLN.id)
+    shipping_zone_pln = shipping_zones[0]
+    shipping_zone_pln_id = graphene.Node.to_global_id(
+        "ShippingZone", shipping_zone_pln.id
+    )
+    variables = {"filter": {"channels": [channel_id]}}
+    # when
+    response = staff_api_client.post_graphql(
+        QUERY_SHIPPING_ZONES_WITH_FILTER,
+        variables=variables,
+        permissions=[permission_manage_shipping],
+    )
+    # then
+    content = get_graphql_content(response)
+    data = content["data"]["shippingZones"]["edges"]
+
+    assert data[0]["node"]["name"] == shipping_zone_pln.name
```

Comment:
I would add an extra check before this line, for the length of data. After all, this test is about making sure we limit the results.